### ISSUE TITLE
Bad Strain Data Getting into the build/Strains with no data

### ISSUE THREAD

**Issue by Paul-Davis** [WormBase/website#5468]
```
WS258:
Primary Source           Connects To Strain
---------------------------------------------------------------------
Condition                RS 2333 WBPaper00041207:PP_Dauer-Exit(12hrs) 
Expr_pattern             None retained.    Expr7771 
Expr_pattern             UL2585 (integrated)
Expr_pattern             cyk-1-GFP-A or B 
Expr_pattern             none retained 
GO_annotation            O1 biovar El Tor
GO_annotation/geneace_Strain                NCTC 8325 
Transgene                OD1421 |OD1422 WBTransgene00019657
Transgene                CLB49 [line 4]
Transgene                CLB49 [line 1] 
caltech_Strain           UL1309, UL1465
```

There are also ~5000 strains where there is no species information 
~3500 from the Transgene class @kyook 
60 from Variation @tuli 
3 from paper @tuli 
Expression pattern @draciti (UL*/TRG*/TH*/VH*/etc.)
Condition class @WenChenWormBase   (WN*/QX*/etc.)

If people could take a look if Species information can/should be added or any additional information?

<pre>
?Strain  Evidence #Evidence
         Genotype UNIQUE Text
         Other_name ?Text #Evidence
         Contains Gene ?Gene XREF Strain
                  Variation ?Variation XREF Strain 
                  Rearrangement ?Rearrangement XREF Strain
                  Clone ?Clone XREF In_strain          
                  Transgene ?Transgene XREF Strain
         Database ?Database ?Database_field ?Text
         Properties Inbreeding_state UNIQUE Selfed Text
                                            Isofemale Text
                                            Multifemale Text
                                            Inbred Text 
                    Outcrossed UNIQUE Text
                    Mutagen Text
                    Strain_history Text
                    Date_first_frozen UNIQUE DateType
                    CGC_received DateType
                    Phenotype ?Phenotype XREF Strain #Phenotype_info
                    Phenotype_not_observed ?Phenotype XREF Not_in_Strain #Phenotype_info //added by Wen to separate Not phenotype from real phenotypes 
         Location ?Laboratory #Lab_Location
         Made_by ?Person //updated from ?text
         Contact ?Person
         Remark ?Text #Evidence
         Reference ?Paper XREF Strain
         Species UNIQUE ?Species // added by krb [021021]
         NCBITaxonomyID UNIQUE Int // Allows for finer grade attachment of NCBItaxon IDs
         Wild_isolate // to identify wild isolates - since genotype is free text
         Isolation GPS UNIQUE Float UNIQUE Float #Position_confidence // Latitude +/-DD.DDDDD Longitude +/-DD.DDDDD
                   Elevation UNIQUE Float #Position_confidence // Elevation above sea level.
                   Elevation UNIQUE Float #Position_confidence // Elevation above sea level.
                   Place UNIQUE ?Text 
                   Landscape UNIQUE Urban_garden
                                    Forest
                                    Wild_grassland
                                    Agricultural_land
                                    Oasis
                                    Rural_garden
                                    Botanical_garden_zoo
                                    Wet_shrubland
                                    Dry_shrubland 
                   Substrate UNIQUE ?Text // e.g. snail, soil, apple, rotting fruit
                   Associated_organisms ?Species
                   Life_stage ?Life_stage
                   Log_size_of_population UNIQUE Float 
                   Sampled_by Text // the person who found the worms
                   Isolated_by ?Person // the person who isolated the worms from the sample
                   Date UNIQUE DateType
                   Picture ?Picture
</pre>

Any new data will probably need to be sent to @tuli for incorporation into the main Strain data set.

**Comment by tuli** [comment]
I wouldn't really call this bad strain data, though it's worth us looking at it. 

For many (most?) of these the Strain name allows the user to find the pertinent data (transgene, variation) having searched WB for the strain. I think that's OK.

Many of the ones connection to Variation data will be populated in a subsequent Strain update from the CGC.

**Comment by kyook** [comment]
Mary Ann will have to let us know what she wants us to do.

Karen

**Comment by Paul-Davis** [comment]
@kyook yes she will but everyone can look at the bad IDs in the 1st section and fix those while a working protocol is devised for populating the empty strains.....thanks for looking into this 👍

**Comment by Paul-Davis** [comment]
Splitting the ticket ad advised.

**Comment by vanaukenk** [comment]
Hi @Paul-Davis and @tuli 

I noticed that there were two strains coming from the GO_annotation class.

- O1 biovar El Tor

- NCTC 8325

These correspond to two bacterial strains that we designate in GO annotations when we use GO terms in the defense response branch of the BP ontology.  

I had thought the names were okay, but please let me know if they need to be updated in any way.  

Thx,
--Kimberly

**Comment by kyook** [comment]
Just to be clear, 

by "bad id" you mean unknown strain name? 

by "fix" you mean attach a species id? 

K

**Comment by Paul-Davis** [comment]
@kyook Not really........I was really only meaning the IDs  like the ones in the table in my pos. Example: "CLB49 [line 4]" where the " [line 4]" has been parsed into an object name.

Unknown strain names as long as valid fall into the other ticket now.

**Comment by tuli** [comment]
@Paul-Davis @vanaukenk @draciti @kyook 
I'll have a look at this tomorrow and get back to you.

**Comment by draciti** [comment]
ok thanks Mary Ann

**Comment by kyook** [comment]
Thanks Mary Ann.

**Comment by kyook** [comment]
Ok, I've corrected the three Transgene bad strain entries. Will be correct in the next upload

**Comment by tuli** [comment]
@draciti 
These are the Exp_pattern errors. 

None retained & None retained : should be removed.
UL2585 (intergrated) : should become UL2585
cyk-1-GFP-A or B : I don't know. This isn't a strain name. You'll need to check the paper to see whether it does refer to an actual strain name. Meanwhile you could move it to a Remark and out of Strain. 

Expr_pattern             None retained.                         Expr7771 
Expr_pattern             UL2585 (integrated)
Expr_pattern             cyk-1-GFP-A or B 
Expr_pattern             none retained

**Comment by tuli** [comment]
@kyook  Thank you. Gold star.

**Comment by tuli** [comment]
@vanaukenk 

These are the GO errors.

GO_annotation            O1 biovar El Tor
GO_annotation/geneace_Strain                NCTC 8325 

From the documentation: Bacterial strain names employ the two or three letter Laboratory/Strain designation, followed by “b”. For example, CBb###. This facilitates distinguishing nematode strains from bacterial strains. Please provide full information on species and relevant genotype of the bacteria.

When we discussed this back in 2014 I went through the existing bacterial strains and added Other_names according to these guidelines e.g. CBX102 http://www.wormbase.org/species/microbacterium_nematophilum/strain/CBX102

So, I think we need to know the origin of these strains. Are they from worm labs?

**Comment by tuli** [comment]
Who curates Condition objects?

**Comment by tuli** [comment]
@Paul-Davis where's the other ticket with the sparsely populated strains?

**Comment by tuli** [comment]
@Paul-Davis "UL1309, UL1465" is a ghost (well they all are) and I don't how to find where it came from. Can you help?

It just needs to be resolved into two strain objects UL1309 and UL1465

**Comment by gw3** [comment]
"UL1309, UL1465" is from the Caltech Strain data ace file.

It is defined in a ghosty line:

Strain : "UL1309, UL1465" -O "2008-10-31_19:44:54_wen"

**Comment by tuli** [comment]
Well yes, I know that much, but I need to know from which object it originates from. Or safe to just ask Wen to delete it?

**Comment by gw3** [comment]
There is no xref to "UL1309, UL1465" that would define it, in the data that Caltech has uploaded.

**Comment by draciti** [comment]
Thanks Mary Ann

none retained fixed:
Expr7771
Expr7783
Expr7785
Expr7786
Expr7724
Expr7776

Expr_pattern UL2585 (integrated) fixed: I put the strain name in the remarks as in OA strain is an autocomplete and UL2585 is not in the list

I cannot find in the expression dump any strain that contains cyk-1-GFP-A or B, do you happen to have the ExprID?

**Comment by gw3** [comment]
'cyk-1-GFP-A or B' is in Expr1200376

**Comment by draciti** [comment]
ok thanks Gary

**Comment by draciti** [comment]
Expr1200376 comes from the TransgeneOme project, it will take a little longer to fix as we need to modify the script that parses their JSON file. Will keep you posted

**Comment by draciti** [comment]
Expr1200376: cyk-1-GFP-A or B. We pinged people at transgeneome site to fix it at their end.

One Strain related question, mainly to Michael P and Mary Ann: it seems that strain OP121 is not in  the nightly gene ace dump but it is in WormBase. What could be the reason?

**Comment by vanaukenk** [comment]
Hi @tuli and @Paul-Davis 

Is the issue with the two strains coming from GO_annotation actually the white spaces in the name?  These seem to be the strain names cited in the C. elegans literature, and in the NCBI taxonomy browser, so I'm not sure what changes we should make, if any.

I double-checked and we have eight Strains cited in GO_annotations with only the last two, OP50 and Db11, containing Other_name entries corresponding to C. elegans labs.  The other strains, e.g. PA14, don't have that information and aren't available from the CGC.

- PA14
- OG1RF 
- O1 biovar El Tor
- NCTC 8355
- SL1344
- V583
- OP50
- Db11

Some relatively quick Textpresso searches show people crediting a range of sources for a given bacterial strain, e.g. other labs (some C. elegans, some not), the ATCC, the NCTC, TIGR, etc., so I'm not sure if it makes sense to try to figure out the exact origin for WB objects.

Thoughts?

**Comment by epaule** [comment]
that would be because OP121 is not in GeneACe, which is the source of the
nightly strain dump.

**Comment by Paul-Davis** [comment]
@vanaukenk white space in the ID is the main issue. Maybe substitute a "_".

**Comment by draciti** [comment]
ok thanks Michael, so where is the info displayed on wormbase coming from?
http://www.wormbase.org/species/all/strain/OP121#02341--10

**Comment by epaule** [comment]
I would think it comes from:
Expr_pattern Expr1200014

**Comment by draciti** [comment]
does this mean that every strain that is in the expr.ace file, in this case

Expr_pattern : Expr1200014 
Pattern	"Data from the TransgeneOme project"
Reporter_gene
Strain	"OP121"
Reference	"WBPaper00041419"
Anatomy_term	"WBbt:0005733" Life_stage "WBls:0000057"
Anatomy_term	"WBbt:0005739" Life_stage "WBls:0000057"
DB_INFO	"TransgeneOme" "gene" "WBGene00007776"
GO_term	"GO:0005730"
Gene	"WBGene00007776"
Reflects_endogenous_expression_of	"WBGene00007776"

will create a new Strain entry on the site?
thx

**Comment by epaule** [comment]
basically, yes. But then most strains should be already at the CGC and
GeneAce, so it would be only for the ones outside that.

**Comment by draciti** [comment]
ok thanks

**Comment by tuli** [comment]
Strains generated from GO_annotation fixed for WS260
